eos-reclaim-swap

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import parted
+import subprocess
+
+class SwapRemover():
+    '''
+    Remove swap partitions from a disk and reclaim their space.
+    '''
+
+    def get_table_str(self, disk):
+        '''
+        Return a text string with the disk partition table.
+        '''
+        cmd = ['fdisk', '-l', disk]
+        out = subprocess.run(cmd, stdout=subprocess.PIPE).stdout
+        return out.decode()
+
+
+    def find_disk(self, partition):
+        '''
+        Return path to a block device representing the disk containing
+        partition.
+        '''
+        p = os.path.realpath(partition)
+        disk = None
+
+        # /dev/sd[a-z][0-9]
+        if (p[:-2] == '/dev/sd'):
+            disk = p[:-1]
+            self.sep = ''
+
+        # /dev/[a-z]+p[0-9], but not /dev/sdp[0-9] (eMMC)
+        elif (p.startswith('/dev/') and p[-2:-1] == 'p'):
+            disk = p[:-2]
+            self.sep = 'p'
+
+        # device-mapped images (dual-boot setups)
+        if (p[-1] == '/dev/mapper/endless-image'):
+            disk = partition[:-1]
+            self.sep = ''
+
+        # for testing with loop devices
+        if (p[:-3] == '/dev/loop'):
```

Comment:
I think you need to assign to `disk` here`

Reply:
As you can probably tell, I ended up not needing to test with loop devices, so I'm going to just drop this, unless you think it could be useful for something else.

Reply:
Dropping it seems good to me.